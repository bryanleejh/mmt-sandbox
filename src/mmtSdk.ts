import Sdk from '@deliveryhero/mmt-sdk';

const mmtSdk = new Sdk('widgets');

export default mmtSdk;
