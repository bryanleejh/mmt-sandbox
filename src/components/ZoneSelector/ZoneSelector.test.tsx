import React from 'react';
import mmtSdk from '@deliveryhero/mmt-sdk';
import { render, fireEvent, cleanup, wait } from '@testing-library/react';
import { MockedProvider } from '@apollo/react-testing';
import { ZoneSelector } from './index';
import { Country } from '../../utils/country';
import { citiesResponseMock, areasResponseMock } from './__mocks__/mock';

const mockCity = 'sg-singapore';

const mockSelectedValues = [{ city: mockCity }];

const defaultComponentProps: any = {
  selectedValues: mockSelectedValues,
  onSelectValues: jest.fn(),
  hasZones: true,
  disabled: false,
};

jest.mock('react-virtualized-auto-sizer', () => {
  const width = 1024;
  const height = 768;
  return ({ children }) => <div>{children({ width, height })}</div>;
});

beforeAll(() => {
  mmtSdk.getActiveCountry = jest.fn().mockReturnValue({ code: 'SG', geid: Country.Singapore });
});

afterEach(cleanup);

describe('ZoneSelector', () => {
  it.skip('should render', async () => {
    const { asFragment } = render(
      <MockedProvider addTypename={false} mocks={[citiesResponseMock, areasResponseMock]}>
        <ZoneSelector {...defaultComponentProps} />
      </MockedProvider>
    );
    expect(asFragment()).toMatchSnapshot();
  });

  it.skip('should render disabled state', async () => {
    const componentPropsWithDisabledState = {
      ...defaultComponentProps,
      disabled: true,
    };
    const { getByTestId } = render(
      <MockedProvider addTypename={false} mocks={[citiesResponseMock, areasResponseMock]}>
        <ZoneSelector {...componentPropsWithDisabledState} disabled />
      </MockedProvider>
    );
    await wait();
    expect(getByTestId('all-cities')).toBeDisabled();
  });

  it.skip('selecting all cities should update all cities', async () => {
    const { getByTestId } = render(
      <MockedProvider addTypename={false} mocks={[citiesResponseMock, areasResponseMock]}>
        <ZoneSelector {...defaultComponentProps} />
      </MockedProvider>
    );
    await wait();
    const allCitiesCheckbox = getByTestId('all-cities');
    expect(allCitiesCheckbox).toBeInTheDocument();
    expect(allCitiesCheckbox.checked).toEqual(false);
    fireEvent.click(allCitiesCheckbox);
    expect(allCitiesCheckbox.checked).toEqual(true);
    fireEvent.click(allCitiesCheckbox);
    expect(allCitiesCheckbox.checked).toEqual(true);
  });

  it.skip('should select city', async () => {
    const { getByTestId } = render(
      <MockedProvider addTypename={false} mocks={[citiesResponseMock, citiesResponseMock]}>
        <ZoneSelector {...defaultComponentProps} />
      </MockedProvider>
    );
    await wait();
    await wait();
    await wait();
    await wait();
    await wait();
    const cityCheckbox = getByTestId('checkbox-singapore');
    expect(cityCheckbox).toBeInTheDocument();
    expect(cityCheckbox.checked).toEqual(true);
    fireEvent.click(cityCheckbox);
    expect(cityCheckbox.checked).toEqual(false);
    fireEvent.click(cityCheckbox);
    expect(cityCheckbox.checked).toEqual(true);
  });

  it.skip('selecting city should render regions', async () => {
    const setSelectCityMock = jest.fn();
    const componentPropsWithCity = {
      ...defaultComponentProps,
      api: {
        ...defaultComponentProps.api,
        selectedCity: 'sg-singapore',
        setSelectedCity: setSelectCityMock,
      },
    };
    const { getByTestId, asFragment } = render(
      <MockedProvider addTypename={false} mocks={[citiesResponseMock, areasResponseMock]}>
        <ZoneSelector {...componentPropsWithCity} />
      </MockedProvider>
    );
    await wait();
    const citySelector = getByTestId('select-singapore');
    expect(citySelector).toBeInTheDocument();
    fireEvent.click(citySelector);
    expect(setSelectCityMock).toBeCalled();
    expect(setSelectCityMock).toBeCalledWith('sg-singapore');
    expect(asFragment()).toMatchSnapshot();
  });

  it.skip('selecting all regions should update all regions', async () => {
    const componentPropsWithCity = {
      ...defaultComponentProps,
      api: {
        ...defaultComponentProps.api,
        selectedCity: 'sg-singapore',
      },
    };
    const { getByTestId } = render(
      <MockedProvider addTypename={false} mocks={[citiesResponseMock, areasResponseMock]}>
        <ZoneSelector {...componentPropsWithCity} />
      </MockedProvider>
    );
    await wait();
    const allRegionCheckbox = getByTestId('all-regions');
    expect(allRegionCheckbox).toBeInTheDocument();
    expect(allRegionCheckbox.checked).toEqual(true);
    fireEvent.click(allRegionCheckbox);
    expect(allRegionCheckbox.checked).toEqual(false);
    fireEvent.click(allRegionCheckbox);
    expect(allRegionCheckbox.checked).toEqual(true);
  });

  it.skip('should select regions', async () => {
    const componentPropsWithCity = {
      ...defaultComponentProps,
      api: {
        ...defaultComponentProps.api,
        selectedCity: 'sg-singapore',
      },
    };
    const { getByTestId } = render(
      <MockedProvider addTypename={false} mocks={[citiesResponseMock, areasResponseMock]}>
        <ZoneSelector {...componentPropsWithCity} />
      </MockedProvider>
    );
    await wait();
    const regionCheckbox = getByTestId('checkbox-central');
    expect(regionCheckbox).toBeInTheDocument();
    expect(regionCheckbox.checked).toEqual(true);
    fireEvent.click(regionCheckbox);
    expect(regionCheckbox.checked).toEqual(false);
    fireEvent.click(regionCheckbox);
    expect(regionCheckbox.checked).toEqual(true);
  });

  it.skip('selecting region should render districts', async () => {
    const componentPropsWithCity = {
      ...defaultComponentProps,
      api: {
        ...defaultComponentProps.api,
        selectedCity: 'sg-singapore',
      },
    };
    const { getByTestId, asFragment } = render(
      <MockedProvider addTypename={false} mocks={[citiesResponseMock, areasResponseMock]}>
        <ZoneSelector {...componentPropsWithCity} />
      </MockedProvider>
    );
    await wait();
    const regionSelector = getByTestId('select-central');
    expect(regionSelector).toBeInTheDocument();
    fireEvent.click(regionSelector);
    expect(asFragment()).toMatchSnapshot();
  });

  it.skip('selecting all districts should update all regions', async () => {
    const componentPropsWithCity = {
      ...defaultComponentProps,
      api: {
        ...defaultComponentProps.api,
        selectedCity: 'sg-singapore',
      },
    };
    const { getByTestId } = render(
      <MockedProvider addTypename={false} mocks={[citiesResponseMock, areasResponseMock]}>
        <ZoneSelector {...componentPropsWithCity} />
      </MockedProvider>
    );
    await wait();
    const regionSelector = getByTestId('select-central');
    expect(regionSelector).toBeInTheDocument();
    fireEvent.click(regionSelector);
    const allDistrictCheckbox = getByTestId('all-districts');
    expect(allDistrictCheckbox).toBeInTheDocument();
    expect(allDistrictCheckbox.checked).toEqual(true);
    fireEvent.click(allDistrictCheckbox);
    expect(allDistrictCheckbox.checked).toEqual(false);
    fireEvent.click(allDistrictCheckbox);
    expect(allDistrictCheckbox.checked).toEqual(true);
  });

  it.skip('should select district', async () => {
    const componentPropsWithCity = {
      ...defaultComponentProps,
      api: {
        ...defaultComponentProps.api,
        selectedCity: 'sg-singapore',
      },
    };
    const { getByTestId } = render(
      <MockedProvider addTypename={false} mocks={[citiesResponseMock, areasResponseMock]}>
        <ZoneSelector {...componentPropsWithCity} />
      </MockedProvider>
    );
    await wait();
    const regionSelector = getByTestId('select-central');
    expect(regionSelector).toBeInTheDocument();
    fireEvent.click(regionSelector);
    const districtCheckbox = getByTestId('checkbox-tanjong-pagar');
    expect(districtCheckbox).toBeInTheDocument();
    expect(districtCheckbox.checked).toEqual(true);
    fireEvent.click(districtCheckbox);
    expect(districtCheckbox.checked).toEqual(false);
    fireEvent.click(districtCheckbox);
    expect(districtCheckbox.checked).toEqual(true);
  });
});
