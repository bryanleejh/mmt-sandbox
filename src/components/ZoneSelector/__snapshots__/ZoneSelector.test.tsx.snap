// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ZoneSelector selecting city should render regions 1`] = `
<DocumentFragment>
  <div
    class="zone-selector"
  >
    <div
      class="zone-panel"
    >
      <div
        class="zone-panel__header"
      >
        <div
          class="zone-panel__header-text"
        >
          Cities
        </div>
        <div
          class="zone-panel__header-input"
        >
          <div
            class="custom-control custom-checkbox"
          >
            <input
              checked=""
              class="custom-control-input"
              data-testid="all-cities"
              id="all-cities"
              type="checkbox"
              value="all-cities"
            />
            <label
              class="zone-panel__header-label custom-control-label"
              data-testid="all-cities-label"
              for="all-cities"
            >
              Select all cities
            </label>
          </div>
        </div>
      </div>
      <div
        class="zone-panel__window"
      >
        <div>
          <div
            style="position: relative; height: 768px; width: 1024px; overflow: auto; will-change: transform; direction: ltr;"
          >
            <div
              style="height: 32px; width: 100%;"
            >
              <div
                class="zone-panel__item zone-panel__item--active"
                style="position: absolute; left: 0px; top: 0px; height: 32px; width: 100%;"
              >
                <div
                  class="custom-control custom-checkbox"
                >
                  <input
                    checked=""
                    class="custom-control-input"
                    data-testid="checkbox-singapore"
                    id="Singapore"
                    type="checkbox"
                    value="sg-singapore"
                  />
                  <label
                    class="custom-control-label"
                    for="Singapore"
                  >
                    Singapore
                  </label>
                </div>
                <svg
                  aria-hidden="true"
                  class="MuiSvgIcon-root MuiSvgIcon-fontSizeSmall"
                  data-testid="select-singapore"
                  focusable="false"
                  style="color: rgb(205, 206, 211);"
                  viewBox="0 0 24 24"
                >
                  <path
                    d="M10 6L8.59 7.41 13.17 12l-4.58 4.59L10 18l6-6z"
                  />
                </svg>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
    <div
      class="zone-panel"
    >
      <div
        class="zone-panel__header"
      >
        <div
          class="zone-panel__header-text"
        >
          Regions
        </div>
        <div
          class="zone-panel__header-input"
        >
          <div
            class="custom-control custom-checkbox"
          >
            <input
              checked=""
              class="custom-control-input"
              data-testid="all-regions"
              id="all-regions"
              type="checkbox"
              value="all-regions"
            />
            <label
              class="zone-panel__header-label custom-control-label"
              data-testid="all-regions-label"
              for="all-regions"
            >
              Select all regions
            </label>
          </div>
        </div>
      </div>
      <div
        class="zone-panel__window"
      >
        <div>
          <div
            style="position: relative; height: 768px; width: 1024px; overflow: auto; will-change: transform; direction: ltr;"
          >
            <div
              style="height: 64px; width: 100%;"
            >
              <div
                class="zone-panel__item "
                style="position: absolute; left: 0px; top: 0px; height: 32px; width: 100%;"
              >
                <div
                  class="custom-control custom-checkbox"
                >
                  <input
                    checked=""
                    class="custom-control-input"
                    data-testid="checkbox-central"
                    id="Central"
                    type="checkbox"
                    value=""
                  />
                  <label
                    class="custom-control-label"
                    for="Central"
                  >
                    Central
                  </label>
                </div>
                <svg
                  aria-hidden="true"
                  class="MuiSvgIcon-root MuiSvgIcon-fontSizeSmall"
                  data-testid="select-central"
                  focusable="false"
                  style="color: rgb(205, 206, 211);"
                  viewBox="0 0 24 24"
                >
                  <path
                    d="M10 6L8.59 7.41 13.17 12l-4.58 4.59L10 18l6-6z"
                  />
                </svg>
              </div>
              <div
                class="zone-panel__item "
                style="position: absolute; left: 0px; top: 32px; height: 32px; width: 100%;"
              >
                <div
                  class="custom-control custom-checkbox"
                >
                  <input
                    checked=""
                    class="custom-control-input"
                    data-testid="checkbox-east"
                    id="East"
                    type="checkbox"
                    value=""
                  />
                  <label
                    class="custom-control-label"
                    for="East"
                  >
                    East
                  </label>
                </div>
                <svg
                  aria-hidden="true"
                  class="MuiSvgIcon-root MuiSvgIcon-fontSizeSmall"
                  data-testid="select-east"
                  focusable="false"
                  style="color: rgb(205, 206, 211);"
                  viewBox="0 0 24 24"
                >
                  <path
                    d="M10 6L8.59 7.41 13.17 12l-4.58 4.59L10 18l6-6z"
                  />
                </svg>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;

exports[`ZoneSelector selecting region should render districts 1`] = `
<DocumentFragment>
  <div
    class="zone-selector"
  >
    <div
      class="zone-panel"
    >
      <div
        class="zone-panel__header"
      >
        <div
          class="zone-panel__header-text"
        >
          Cities
        </div>
        <div
          class="zone-panel__header-input"
        >
          <div
            class="custom-control custom-checkbox"
          >
            <input
              checked=""
              class="custom-control-input"
              data-testid="all-cities"
              id="all-cities"
              type="checkbox"
              value="all-cities"
            />
            <label
              class="zone-panel__header-label custom-control-label"
              data-testid="all-cities-label"
              for="all-cities"
            >
              Select all cities
            </label>
          </div>
        </div>
      </div>
      <div
        class="zone-panel__window"
      >
        <div>
          <div
            style="position: relative; height: 768px; width: 1024px; overflow: auto; will-change: transform; direction: ltr;"
          >
            <div
              style="height: 32px; width: 100%;"
            >
              <div
                class="zone-panel__item zone-panel__item--active"
                style="position: absolute; left: 0px; top: 0px; height: 32px; width: 100%;"
              >
                <div
                  class="custom-control custom-checkbox"
                >
                  <input
                    checked=""
                    class="custom-control-input"
                    data-testid="checkbox-singapore"
                    id="Singapore"
                    type="checkbox"
                    value="sg-singapore"
                  />
                  <label
                    class="custom-control-label"
                    for="Singapore"
                  >
                    Singapore
                  </label>
                </div>
                <svg
                  aria-hidden="true"
                  class="MuiSvgIcon-root MuiSvgIcon-fontSizeSmall"
                  data-testid="select-singapore"
                  focusable="false"
                  style="color: rgb(205, 206, 211);"
                  viewBox="0 0 24 24"
                >
                  <path
                    d="M10 6L8.59 7.41 13.17 12l-4.58 4.59L10 18l6-6z"
                  />
                </svg>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
    <div
      class="zone-panel"
    >
      <div
        class="zone-panel__header"
      >
        <div
          class="zone-panel__header-text"
        >
          Regions
        </div>
        <div
          class="zone-panel__header-input"
        >
          <div
            class="custom-control custom-checkbox"
          >
            <input
              checked=""
              class="custom-control-input"
              data-testid="all-regions"
              id="all-regions"
              type="checkbox"
              value="all-regions"
            />
            <label
              class="zone-panel__header-label custom-control-label"
              data-testid="all-regions-label"
              for="all-regions"
            >
              Select all regions
            </label>
          </div>
        </div>
      </div>
      <div
        class="zone-panel__window"
      >
        <div>
          <div
            style="position: relative; height: 768px; width: 1024px; overflow: auto; will-change: transform; direction: ltr;"
          >
            <div
              style="height: 64px; width: 100%;"
            >
              <div
                class="zone-panel__item zone-panel__item--active"
                style="position: absolute; left: 0px; top: 0px; height: 32px; width: 100%;"
              >
                <div
                  class="custom-control custom-checkbox"
                >
                  <input
                    checked=""
                    class="custom-control-input"
                    data-testid="checkbox-central"
                    id="Central"
                    type="checkbox"
                    value=""
                  />
                  <label
                    class="custom-control-label"
                    for="Central"
                  >
                    Central
                  </label>
                </div>
                <svg
                  aria-hidden="true"
                  class="MuiSvgIcon-root MuiSvgIcon-fontSizeSmall"
                  data-testid="select-central"
                  focusable="false"
                  style="color: rgb(205, 206, 211);"
                  viewBox="0 0 24 24"
                >
                  <path
                    d="M10 6L8.59 7.41 13.17 12l-4.58 4.59L10 18l6-6z"
                  />
                </svg>
              </div>
              <div
                class="zone-panel__item "
                style="position: absolute; left: 0px; top: 32px; height: 32px; width: 100%;"
              >
                <div
                  class="custom-control custom-checkbox"
                >
                  <input
                    checked=""
                    class="custom-control-input"
                    data-testid="checkbox-east"
                    id="East"
                    type="checkbox"
                    value=""
                  />
                  <label
                    class="custom-control-label"
                    for="East"
                  >
                    East
                  </label>
                </div>
                <svg
                  aria-hidden="true"
                  class="MuiSvgIcon-root MuiSvgIcon-fontSizeSmall"
                  data-testid="select-east"
                  focusable="false"
                  style="color: rgb(205, 206, 211);"
                  viewBox="0 0 24 24"
                >
                  <path
                    d="M10 6L8.59 7.41 13.17 12l-4.58 4.59L10 18l6-6z"
                  />
                </svg>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
    <div
      class="zone-panel-fluid"
    >
      <div
        class="zone-panel__header"
      >
        <div
          class="zone-panel__header-text"
        >
          Districts
        </div>
        <div
          class="zone-panel__header-input"
        >
          <div
            class="custom-control custom-checkbox"
          >
            <input
              checked=""
              class="custom-control-input"
              data-testid="all-districts"
              id="all-districts"
              type="checkbox"
              value="all-districts"
            />
            <label
              class="zone-panel__header-label custom-control-label"
              data-testid="all-districts-label"
              for="all-districts"
            >
              Select all districts
            </label>
          </div>
        </div>
      </div>
      <div
        class="zone-panel__window"
      >
        <div>
          <div
            style="position: relative; height: 768px; width: 1024px; overflow: auto; will-change: transform; direction: ltr;"
          >
            <div
              style="height: 64px; width: 100%;"
            >
              <div
                class="zone-panel__item "
                style="position: absolute; left: 0px; top: 0px; height: 32px; width: 100%;"
              >
                <div
                  class="custom-control custom-checkbox"
                >
                  <input
                    checked=""
                    class="custom-control-input"
                    data-testid="checkbox-tanjong-pagar"
                    id="Tanjong Pagar"
                    type="checkbox"
                    value=""
                  />
                  <label
                    class="custom-control-label"
                    for="Tanjong Pagar"
                  >
                    Tanjong Pagar
                  </label>
                </div>
                <svg
                  aria-hidden="true"
                  class="MuiSvgIcon-root MuiSvgIcon-fontSizeSmall"
                  data-testid="select-tanjong-pagar"
                  focusable="false"
                  style="color: rgb(205, 206, 211);"
                  viewBox="0 0 24 24"
                >
                  <path
                    d="M10 6L8.59 7.41 13.17 12l-4.58 4.59L10 18l6-6z"
                  />
                </svg>
              </div>
              <div
                class="zone-panel__item "
                style="position: absolute; left: 0px; top: 32px; height: 32px; width: 100%;"
              >
                <div
                  class="custom-control custom-checkbox"
                >
                  <input
                    checked=""
                    class="custom-control-input"
                    data-testid="checkbox-chinatown"
                    id="Chinatown"
                    type="checkbox"
                    value=""
                  />
                  <label
                    class="custom-control-label"
                    for="Chinatown"
                  >
                    Chinatown
                  </label>
                </div>
                <svg
                  aria-hidden="true"
                  class="MuiSvgIcon-root MuiSvgIcon-fontSizeSmall"
                  data-testid="select-chinatown"
                  focusable="false"
                  style="color: rgb(205, 206, 211);"
                  viewBox="0 0 24 24"
                >
                  <path
                    d="M10 6L8.59 7.41 13.17 12l-4.58 4.59L10 18l6-6z"
                  />
                </svg>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;

exports[`ZoneSelector should render 1`] = `
<DocumentFragment>
  <div
    class="zone-selector"
  >
    <div
      class="loading--component"
    >
      <div
        class="spinner-border"
      />
    </div>
    <div
      class="zone-panel"
    >
      <div
        class="zone-panel__header"
      >
        <div
          class="zone-panel__header-text"
        >
          Cities
        </div>
        <div
          class="zone-panel__header-input"
        >
          <div
            class="custom-control custom-checkbox"
          >
            <input
              checked=""
              class="custom-control-input"
              data-testid="all-cities"
              id="all-cities"
              type="checkbox"
              value="all-cities"
            />
            <label
              class="zone-panel__header-label custom-control-label"
              data-testid="all-cities-label"
              for="all-cities"
            >
              Select all cities
            </label>
          </div>
        </div>
      </div>
      <div
        class="zone-panel__window"
      >
        <div>
          <div
            style="position: relative; height: 768px; width: 1024px; overflow: auto; will-change: transform; direction: ltr;"
          >
            <div
              style="height: 0px; width: 100%;"
            />
          </div>
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
